### DISCUSSION TITLE
[filesystem] toggle gitignored and toggle hidden

### DISCUSSION THREAD

**Discussion by MunifTanjim** [nvim-neo-tree/neo-tree.nvim#259] [Q&A] [answered]
How to toggle git ignored files and hidden files separately (using two different keybindings)?

- **git ignored files**: specified by `.gitignore` file
- **hidden files**: files starting with a `.` character

**Comment by cseickel** [comment, marked as answer]
You could create custom mappings to toggle them individually, but there is no way (currently) to combine this with the ability to mark hidden files in a different highlight.

```lua
        mappings = {
          ["G"] = function (state)
            state.filtered_items.visible = false
            state.filtered_items.hide_gitignored = not state.filtered_items.hide_gitignored
            require("neo-tree.sources.manager").refresh("filesystem")
          end,
          ["H"] = function (state)
            state.filtered_items.visible = false
            state.filtered_items.hide_dotfiles = not state.filtered_items.hide_dotfiles
            require("neo-tree.sources.manager").refresh("filesystem")
          end,
```

If you want to also see them grayed out when they are not hidden, that would require enhancing the functionality in the internals of Neo-tree.
